kadmin/server.c

```diff
@@ -125,6 +127,10 @@ kadmind_dispatch(void *kadm_handlep, krb5_boolean initial,
 	ret = kadm5_get_principal(kadm_handlep, princ, &ent, mask);
 	krb5_storage_free(sp);
 	sp = krb5_storage_emem();
+        if (sp == NULL) {
+            ret = krb5_enomem(contextp->context);
```

Comment:
Also `krb5_free_principal(contextp->context, princ);`.  But I've been meaning to move that to the `fail` label, and initialize `princ` to `NULL` at the top.